File: app/src/main/java/com/molihua/hlbmerge/entity/CacheSrc.java
package com.molihua.hlbmerge.entity;

public class CacheSrc<T> {
    private T audio;
    private T video;
    private T json;
    private T danmaku;

    public T getAudio() {
        return audio;
    }

    public CacheSrc setAudio(T audio) {
        this.audio = audio;
        return this;
    }

    public T getVideo() {
        return video;
    }

    public CacheSrc setVideo(T video) {
        this.video = video;
        return this;
    }

    public T getJson() {
        return json;
    }

    public CacheSrc setJson(T json) {
        this.json = json;
        return this;
    }

    public T getDanmaku() {
        return danmaku;
    }

    public CacheSrc setDanmaku(T danmaku) {
        this.danmaku = danmaku;
        return this;
    }
}

File: app/src/main/java/com/molihua/hlbmerge/entity/CacheSrc.kt
package com.molihua.hlbmerge.entity

class CacheSrc<T> {
    var audio: T? = null
        private set
    var video: T? = null
        private set
    var json: T? = null
        private set
    var danmaku: T? = null
        private set

    fun setAudio(audio: T): CacheSrc<T> {
        this.audio = audio
        return this
    }

    fun setVideo(video: T): CacheSrc<T> {
        this.video = video
        return this
    }

    fun setJson(json: T): CacheSrc<T> {
        this.json = json
        return this
    }

    fun setDanmaku(danmaku: T): CacheSrc<T> {
        this.danmaku = danmaku
        return this
    }
}